graphql/resolvers/addComment.ts

```diff
@@ -11,6 +11,129 @@ import {
   PRIMARY_COLOR_HEX,
   PROFILE_URL
 } from '../../constants'
+import { Challenge, Comment, Lesson, Submission, User } from '@prisma/client'
+
+type CommentWithAuthorAndSubmission = Comment & {
+  author: {
+    id: number
+    username: string
+    discordId: string | null
+    name: string
+  }
+  submission: Submission & {
+    user: { id: number; username: string; discordId: string | null }
+    lesson: Lesson
+    challenge: Challenge
+  }
+}
+
+type SendNotificationsParams = {
+  line: InputMaybe<number> | undefined
+  fileName: InputMaybe<string> | undefined
+  submissionId: number
+  comment: CommentWithAuthorAndSubmission
+}
+
+const sendNotifications = async ({
+  line,
+  submissionId,
+  fileName,
+  comment
+}: SendNotificationsParams) => {
+  const restOfComments = await prisma.comment.findMany({
+    where: {
+      line,
+      submissionId,
+      fileName
+    },
+    include: {
+      author: true
+    }
+  })
+
+  const [uniqueParticipants] = restOfComments?.reduce(
+    (
+      acc: [(Comment & { author: User })[], { [key: string]: boolean }],
+      e: Comment & { author: User }
+    ) => {
+      const [arr, map] = acc
+
+      if (!map[e.authorId]) {
+        arr.push(e)
+        map[e.authorId] = true
+      }
+
+      return acc
+    },
+    [[], {}]
+  )
+
+  const { submission, author } = comment
+
+  const notificationEmbed = (isSubmissionOwner: boolean): APIEmbed => {
+    const params = {
+      title: isSubmissionOwner
```

Comment:
Is this doing anything ? submission owner does not get notified of the comments they make so this will never be true.

Reply:
I think it should've been "forSubmissionOwner" instead. If the flag is true, a customized embed message is returned to the submission owner. It's only true when someone, such as a reviewer (someone who isn't the owner of the submission), leaves a comment on their submission.

See line 114 when the comment author isn't the submission owner.

Reply:
I see got it, yeah I think `forSubmissionOwner` would make the purpose clearer then sure you can go with that